### ISSUE TITLE
library collisions on centos7 for emacs

### ISSUE THREAD

**Issue by jeanconn** [sot/skare3#121]
```
luke-v: ska3
ska3-luke-v: emacs
emacs: /proj/sot/ska3/flight/lib/libtiff.so.5: no version information available (required by emacs)
emacs: symbol lookup error: /lib64/libpango-1.0.so.0: undefined symbol: g_log_structured_standard
```

I'm guessing this isn't a problem on CentOS-less-than-7 because /usr/bin/emacs is older and wants, say, libtiff.so.3 and does not collide.

**Comment by javierggt** [comment]
was this fixed? How does this affect skare3?

**Comment by taldcroft** [comment]
From my testing:
- Fixed on kady (CentOS-7)
- Not fixed on cheru (CentOS-8)
```
ska3-cheru% which python
/proj/sot/ska3/flight/bin/python
ska3-cheru% emacs -nw
emacs: /proj/sot/ska3/flight/lib/libtiff.so.5: no version information available (required by emacs)
emacs: /proj/sot/ska3/flight/lib/libuuid.so.1: no version information available (required by /lib64/libSM.so.6)
emacs: symbol lookup error: /lib64/libwebkit2gtk-4.0.so.37: undefined symbol: gst_gl_display_egl_new_with_egl_display
```

**Comment by javierggt** [comment]
so basically it seems that the possible "solutions" are:
- not to use emacs within ska3 environment
- install emacs within the ska environment (no ideal, let's say)
- see if there is a set of versions that uses compatible library versions

Did I get it right?

**Comment by taldcroft** [comment]
You forgot option 4, what I currently do on HEAD:
```
kady$ alias emacs
alias emacs='env LD_LIBRARY_PATH= /bin/emacs'
```

**Comment by taldcroft** [comment]
Can we put a light launcher in ska3 bin that does the same?

**Comment by javierggt** [comment]
Ah, that is a solution.

Can we consider this solved? Or do we want to do something?

I think we can put a light launcher.

**Comment by javierggt** [comment]
Would be nice to group that with changes to flt_env (#73, #82) and ska_version (#143)

**Comment by taldcroft** [comment]
Are you in favor of option 4 (with a documentation update, e.g. in the Configuration Hints of Ska3 for Users) or the light launcher option 5?  Doing option 4 is a 2-minute thing, option 5 is a bit longer but then it just works automatically for everyone. Though I'm not sure how to do option 5 in the general case where you don't know where `emacs` lives in advance (e.g. standalone).

**Comment by taldcroft** [comment]
Option 6: do we still actually need to set `LD_LIBRARY_PATH` with `/proj/sot/ska3/flight/lib`? Maybe we do, but I just don't know.

**Comment by javierggt** [comment]
> Though I'm not sure how to do option 5 in the general case where you don't know where emacs lives in advance (e.g. standalone).

Wouldn't something like this work?
```
#!/usr/bin/env bash
env --unset=LD_LIBRARY_PATH emacs $@
```

**Comment by javierggt** [comment]
also the alias:
```
alias emacs="env --unset=LD_LIBRARY_PATH emacs"
```

**Comment by taldcroft** [comment]
I'm not sure if that will recurse if that file is named `emacs` and is first in the path. I would think yes, but worth trying.

**Comment by javierggt** [comment]
I noticed that emacs works in kady. It still issues the warnings about libtiff and libuuid. So the main error might be the undefined symbol. I googled that, and for example I found this: https://bugzilla.redhat.com/show_bug.cgi?id=1499466

Perhaps that's a thing that could be solved in the system itself?

**Comment by jeanconn** [comment]
Regarding "Option 6: do we still actually need to set LD_LIBRARY_PATH with /proj/sot/ska3/flight/lib? Maybe we do, but I just don't know.", I think the question is both that, and do we need to be in a sourced ska3 for most cmdline work?  Maybe we could just conda activate our own root environment instead of using the current flt_envs.

**Comment by taldcroft** [comment]
I think that getting rid of the sourced environment in our linux production environments could be going too far. There is just so much heritage in expecting certain env vars to be set e.g. `$SKA_DATA`, even if now we make new code differently. Anyway, just for kicks I'm doing this on `kady`:
```
export LD_LIBRARY_PATH=/proj/sot/ska3/flight/pgplot:/soft/SYBASE16.0/OCS-16_0/lib
run_testr
```

**Comment by jeanconn** [comment]
Right, I just meant for cmdline work.  I figure for production tasks, `skare` or being in a src'd env to test a production script makes sense, but I figure you don't source into ska3 with flt_envs on your mac, so you might not need to for work on a linux machine (for most things) either.

**Comment by jeanconn** [comment]
Also, the emacs lib colliision only came up for me on cheru because I was trying to share a window with you two during the upgrade and hadn't used cheru for much else.

**Comment by taldcroft** [comment]
FWIW ska_testr passed on kady (except expected fail of mica) when started from an env where LD_LIBRARY_PATH does not have the ska3 lib dir.

I'd rather not make a distinction between cmdline work and production processing on HEAD, unless we really need to. Our systems are confusing enough already.

**Comment by jeanconn** [comment]
I just though we had, sort of, when you'd established that for new code, we
just need to be in the conda environment and have SKA set.

**Comment by jeanconn** [comment]
Closed by #855
